python/cudf/cudf/utils/dtypes.py

```diff
@@ -235,3 +235,35 @@ def min_numeric_column_type(x):
         return np.promote_types(max_bound_dtype, min_bound_dtype)
 
     return x.dtype
+
+
+def check_upcast_unsupported_dtype(dtype):
+    from cudf._lib.types import np_to_cudf_types
+
+    if is_categorical_dtype(dtype):
+        return dtype
+    dtype = np.dtype(dtype)
+    if dtype in np_to_cudf_types:
+        return dtype
+
+    if dtype.kind in ("u", "i"):
+        supported_i_types = [
+            np.dtype("int8"),
+            np.dtype("int16"),
+            np.dtype("int32"),
+            np.dtype("int64"),
+        ]
+        for supported_type in supported_i_types:
```

Comment:
This looping looks expensive. Can we just build a mapping? I.E. `uint8` --> `int16`, `uint16` --> `int32`, `uint32` --> `int64`, `uint64` --> `int64` (may need to check for overflow, sounds expensive).

Reply:
@kkraus14 Added a map to handle this but removed `uint64` to `int64` mapping as say if we have `18446744073709551615`(max value in `uint64`) the numpy typecast seems to be changing to -1's so will either have to scan the entire array before type-casting for such out of `int64` range values or tally the -1's count of which both seem to be equally expensive. 

But if we want to have this type-cast happen we could just print a warning message that data could be lost due to the downcast happening and ignore validating the data for lost values.

Reply:
I'm in support of a warning when typecasting `uint64` that we won't protect against overflow. That sounds like a good tradeoff.

Reply:
Added support for `uint64` to `int64` cast with a warning message.